src/uct/api/uct.h

```diff
@@ -426,6 +418,33 @@ typedef enum uct_atomic_op {
  */
 
 
+/**
+ * @defgroup UCT_RESOURCE_IFACE_EVENT_CAP   UCT interface for asynchronous event capabilities
+ * @ingroup UCT_RESOURCE
+ *
+ * @brief List of capabilities supported by UCT iface event API
+ *
+ * The definition list presents a full list of operations and capabilities
+ * supported by UCT iface event.
+ * @{
+   */
+        /* Event types */
+#define UCT_IFACE_FLAG_EVENT_SEND_COMP UCS_BIT(0) /**< Event notification of send completion is
+                                                       supported */
+#define UCT_IFACE_FLAG_EVENT_RECV      UCS_BIT(1) /**< Event notification of tag and active message
+                                                       receive is supported */
+#define UCT_IFACE_FLAG_EVENT_RECV_SIG  UCS_BIT(2) /**< Event notification of signaled tag and active
```

Comment:
So, it receive is just completion on receive queue ? Does SIG represent completion on RDMA with immediate ? How we can communicate to upper layer that some type of event is not supported ? think about other interconnect that has no support for events (shared memory) or supports only some events (ugni)

Reply:
@shamisp those flags were in the UCT before this PR. They weren't introduced in this PR, they have been moved to another place only
`UCT_IFACE_FLAG_EVENT_RECV_SIG` is supported only by MM, it means that receive will be signaled if AM Bcopy with `UCT_SEND_FLAG_SIGNALED` flag. It is needed to make first send signaled in UCP (proxy EP)
IB supports `UCT_IFACE_FLAG_EVENT_RECV`, when each receive can generate Event if iface "armed" for them. in IB we support first signaled event thru solicited event

Reply:
Got it. IMHO it is poorly documented.

Reply:
> Got it. IMHO it is poorly documented.

we will consider a fix for this in the further PRs